pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/exchange/BlockExchange.java

```diff
@@ -110,10 +129,16 @@ public boolean send(TransferableBlock block)
   }
 
   protected void sendBlock(SendingMailbox sendingMailbox, TransferableBlock block)
-      throws Exception {
+      throws IOException, TimeoutException {
+    if (LOGGER.isTraceEnabled()) {
+      LOGGER.trace("Sending block: {} {} to {}", block.getType(), System.identityHashCode(block), sendingMailbox);
```

Comment:
Why `identityHashCode` and not the `toString` representation?

Reply:
These logs are useful when debuging intergration tests. In production code it wouldn't be useful because too many logs would be printed.

The `toString` representation is not very useful. It includes the type of the block and the number of rows. That is not identifying anything. It is very easy to have two blocks distinct blocks in the stage that have the same type and number of rows. This is why we use `identityHashCode` in some logs (example, `BlockingMultiStreamConsumer`).

In local debug tests, it is very useful to have this id to track how a block is being processed by the engine without being fooled by other blocks with the same number of rows.